tests/integration/Owncloud/OcisSdkPhp/OcisTest.php

```diff
@@ -139,14 +139,19 @@ public function testCreateDriveInvalidQuota(int $quota): void
         $this->assertCount($countDrivesAtStart, $ocis->listMyDrives());
     }
 
+    /**
+     * @return void
+     */
     public function testGetGroups(): void
     {
         $token = $this->getAccessToken('admin', 'admin');
         $ocis = new Ocis($this->ocisUrl, $token, ['verify' => false]);
+        $ocis->createGroup("philosophy-haters", "sss");
+        $ocis->createGroup("physics-lovers", "sss");
```

Comment:
We should move this to the `setUp` function, or have the `createGroup` calls in every test, otherwise we cannot run some tests independently of others and the order of tests cannot be changed.
In future we need to be able to delete groups, and then we can have a proper cleanuo

Reply:
If this is kept in `setup` , then ocis should be used as precondition for test since ocis ffirst need to get initialized. Also, we haven't implemented deletegroup, creating group in every step would mean that every assertion steps would have more group related information in it and it would fail when we implement delete group. We can use `createGroup` in every step once we implement the delete group.